File: ts/src/emitter/stats/format-label.ts
import type { VisualNode } from "../../visual-graph/visual-node.js";

export function formatLabel(path: string, n: VisualNode): string {
  const prefix = n.unused ? "unused " : "";
  return `${path}:${n.line} ${prefix}${n.name}`;
}

File: crates/unsnarl-emitter-stats/src/format_label.rs
//! Formats the third TSV column of a stats row: `path:line [unused ]name`.
//!
//! Mirrors `ts/src/emitter/stats/format-label.ts`. The leading
//! `path:line` is what makes the row clickable in editors that
//! pick up jump-to-source targets; the `unused ` prefix surfaces
//! the IR-level "unused" annotation already attached by the
//! analyzer.

use unsnarl_visual_graph::visual_node::VisualNode;

pub fn format_label(path: &str, n: &VisualNode) -> String {
    let prefix = if n.unused() { "unused " } else { "" };
    format!(
        "{path}:{line} {prefix}{name}",
        line = n.line(),
        name = n.name()
    )
}

#[cfg(test)]
#[path = "format_label_test.rs"]
mod format_label_test;
